Initial Commit

sshurie opened the pull request:
Assignment 4: Methods

sshurie committed:
Initial Commit

Assignment 4: Methods

Paa1006 commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+       
+      //While loop: run this menu as long as the user does not enter 6 on the main menu
+        while (input != 6) {
+            
```
It doesn't really effect the operation of the code, but you could have used a Do while loop here since the code should always run at least once. No need to pretest to loop. This works fine though.

ninjapanzer replied:
I agree, but you would be surprised how unusual `do{}while()` loops are in the wild.

DTkach commented:
Really good documentation. I kinda forgot to add those comment blurbs before each function in mine...whoops!

Also, I agree with gary. Adding the menu to a method is a smart idea!

ninjapanzer commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+     * score, percentage and letter grade
+     * @param keyboard Scanner keyboard reads user input 
+     */
+    public static void gradeCalc(Scanner keyboard) {
```
This is a fine method I like the reuses of Scanner. When we do objects this will be a pattern we will extract a lot. :) 

The feedback I will give is that this single method is rather large, like its own program really. I know that is exactly what I asked you to do but we are going to introduce the way to resolve this issue tonight.

sshurie replied:
@garyanewsome , @Hallbomb @DTkach thank you for your feedback! I left it out originally but then I thought it looked somewhat awkward left out like that while everything else had it's own method :)

sshurie replied:
@DTkach Thank you! I forget about the documentation sometimes, too. It isn't quite a habit for me yet. Usually I go back through after I'm finished coding everything and then add the comments but that can seem a bit tedious. I guess we'll just have to remember, when you start a method, close it and also add your comments right away!

sshurie replied:
@ztmurphy21 Thank you for pointing that out. I fixed the slot machine game but it isn't up quite yet. I will upload it whenever I get home from work.

ninjapanzer commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+      //Chapter 4 Exercise 22: Slot Machine Simulation (page 267)
+      //Objective: Create a program that simulates a slot machine
+        Integer betAmt, winnings;
+        Random spin = new Random(); //random object generates new random number
```
Because you were nice enough to introduce and use two system object in this method I will probably use this code as an example of building classes tonight. :shipit:

ninjapanzer commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+          "Adventure Time,", "Come on, grab your friends", "We'll go to" , 
+            "very distant lands", "With Jake the Dog", "And Finn the Human",
+            "The fun will never end,", "It's Adventure Time!"
+        };    
```
:heart: :yellow_heart: :green_heart: :blue_heart: :green_heart: :yellow_heart:  :heart:

ninjapanzer commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+            */
+            for(int i=0; i < numLoops; i++) {
+                print("\t\t" + lyrics[i] +"\n");
+            } 
```
Perfect. When we do arrays I will cover list comprehensions and we will learn a new kind of loop that is perfect for this case. :+1:

sshurie replied:
Is it a for-each loop? We were learning about those in Visual Basic. Does Java have something like that?

ninjapanzer replied:
In Java we call it an iterator driven loop but there is a shortcut that looks like for-each

``` java
Arraylist<String> thingCollection = new Arraylist<String>(Arrays.asList("one", "two", "three"));
for (Thing variable : thingCollection){
   System.out.println("Next Thing is "+ variable)
}
```

Output:
Next Thing is one
Next Thing is two
Next Thing is three

ninjapanzer commented:
```
run:
SLOT MACHINE
    Place Your Bet (dollar amount [Integer]): 100

    [ Plums ] [ Plums ] [ Plums ]

    Congratulations! 3 Matching!
    You Won: $300
To play again, enter 4 on the menu
BUILD SUCCESSFUL (total time: 3 seconds)
```

:heart:

ninjapanzer commented on arevyn24-assignment-4/src/arevyn24/assignment/pkg4/Assignment4Methods.java:
```diff
@@ -0,0 +1,329 @@
+       * slotValue() method assigns text value to each slot
+       */
+        for(int i = 0; i < randoms.length; i++) {
+            randoms[i] = spin.nextInt(6);
```
Pseudo random numbers are hard. `.nextInt(6)` in this context might return the same number every time.

You have two choices:
1. Create a new instance of Random for each loop `randoms[i] = (new Random()).nextInt(6);
2. get a random window of the seedfile and you can skip the loop assignment for randoms `randoms = spin.ints(1,7).limit(3).toArray();`

That second one is a little heavier. I think randoms are fun personally so maybe we will spend a little more time on it in class later this semester.

Hallbomb commented:
I agree with Gary as well, adding it to a method is a nice touch.

sshurie committed:
Loops and Other Fixes

changed to do-while loop as suggested by @Paa1006 , Also fixed the slot
machine outcomes, so no more jackpots 100% of the time @ztmurphy21
@ninjapanzer
